clang/Util.d

```diff
@@ -80,14 +80,32 @@ mixin template CX ()
 }
 
 extern (C) int mkstemps(char*, int);
+extern (C) char* mkdtemp(char*);
 extern (C) int close(int);
 
 class NamedTempFileException : object.Exception
 {
-    this (string message, string file = __FILE__, size_t line = __LINE__)
+    this (string path, string file = __FILE__, size_t line = __LINE__)
     {
-        super(message, file, line);
+        this.path = path;
+        super(format("Cannot create temporary file \"%s\".", path), file, line);
     }
+
+    public string path;
```

Comment:
Please always put instance variables at the top, before all methods and constructors. `public` is also not necessary. Set it to const or immutable as well to make it read only.